### ISSUE TITLE
Unable to replay http requests on laravel

### ISSUE THREAD

**Issue by kilwir76** [dunglas/frankenphp#538]
### What happened?

When you cause a bug under laravel with frankenphp, it's impossible to replay call api via postman or insonmia or any other such application.

There is no response from laravel.

The server remains frozen. 

I don't use octane, to be precise.

### Build Type

Custom (tell us more in the description)

### Worker Mode

No

### Operating System

GNU/Linux

### CPU Architecture

x86_64

**Comment by dunglas** [comment]
Would you be able to provide us a reproducer?

**Comment by MakFly** [comment]
Yes of course . I'm doing this tonight

**Comment by jerevive** [comment]
this is my result 

```php
./frankenphp run
2024/02/02 03:08:31.296 INFO    using adjacent Caddyfile
2024/02/02 03:08:31.299 INFO    admin   admin endpoint started  {"address": "localhost:2019", "enforce_origin": false, "origins": ["//localhost:2019", "//[::1]:2019", "//127.0.0.1:2019"]}
2024/02/02 03:08:31.299 INFO    http.auto_https server is listening only on the HTTPS port but has no TLS connection policies; adding one to enable TLS   {"server_name": "srv0", "https_port": 443}2024/02/02 03:08:31.299 INFO    http.auto_https enabling automatic HTTP->HTTPS redirects        {"server_name": "srv0"}
2024/02/02 03:08:31.299 INFO    tls.cache.maintenance   started background certificate maintenance{"cache": "0xc0001d6180"}
2024/02/02 03:08:31.302 INFO    FrankenPHP started 🐘    {"php_version": "8.3.2"}
2024/02/02 03:08:31.307 INFO    tls     cleaning storage unit   {"storage": "FileStorage:/root/.local/share/caddy"}
2024/02/02 03:08:31.307 INFO    tls     finished cleaning storage units
2024/02/02 03:08:31.328 WARN    pki.ca.local    installing root certificate (you might be prompted for password)  {"path": "storage:pki/authorities/local/root.crt"}
2024/02/02 03:08:31.328 INFO    not NSS security databases found
2024/02/02 03:08:31.328 INFO    define JAVA_HOME environment variable to use the Java trust
2024/02/02 03:08:31.780 INFO    certificate installed properly in linux trusts
2024/02/02 03:08:31.781 INFO    http    enabling HTTP/3 listener        {"addr": ":443"}
2024/02/02 03:08:31.781 INFO    http.log        server running  {"name": "srv0", "protocols": ["h1", "h2", "h3"]}
2024/02/02 03:08:31.781 INFO    http.log        server running  {"name": "remaining_auto_https_redirects", "protocols": ["h1", "h2", "h3"]}
2024/02/02 03:08:31.781 INFO    http    enabling automatic TLS certificate management   {"domains": ["localhost"]}
2024/02/02 03:08:31.782 INFO    autosaved config (load with --resume flag)      {"file": "/root/.config/caddy/autosave.json"}
2024/02/02 03:08:31.782 INFO    serving initial configuration
2024/02/02 03:08:31.783 INFO    tls.obtain      acquiring lock  {"identifier": "localhost"}
2024/02/02 03:08:31.795 INFO    tls.obtain      lock acquired   {"identifier": "localhost"}
2024/02/02 03:08:31.795 INFO    tls.obtain      obtaining certificate   {"identifier": "localhost"}
2024/02/02 03:08:31.796 INFO    tls.obtain      certificate obtained successfully       {"identifier": "localhost"}
2024/02/02 03:08:31.796 INFO    tls.obtain      releasing lock  {"identifier": "localhost"}
2024/02/02 03:08:31.796 WARN    tls     stapling OCSP   {"error": "no OCSP stapling for [localhost]: no OCSP server specified in certificate", "identifiers": ["localhost"]}
2024/02/02 03:08:52.601 INFO    admin.api       received request        {"method": "GET", "host": "127.0.0.1:2019", "uri": "/", "remote_ip": "127.0.0.1", "remote_port": "43706", "headers": {"Accept":["*/*"],"User-Agent":["curl/7.29.0"]}}
2024/02/02 03:19:32.570 INFO    admin.api       received request        {"method": "GET", "host": "127.0.0.1:2019", "uri": "/", "remote_ip": "127.0.0.1", "remote_port": "43708", "headers": {"Accept":["*/*"],"User-Agent":["curl/7.29.0"]}}
2024/02/02 03:20:43.000 INFO    admin.api       received request        {"method": "GET", "host": "localhost:2019", "uri": "/", "remote_ip": "127.0.0.1", "remote_port": "43712", "headers": {"Accept":["*/*"],"User-Agent":["curl/7.29.0"]}}

```

```
[root@localhost ~]# curl localhost:2019
404 page not found
```

**Comment by jerevive** [comment]
this is my result 


```php
docker run -p 80:80 -p 443:443 -p 443:443/udp -v $PWD:/app dungl
as/frankenphp
{"level":"info","ts":1706845343.954082,"msg":"using provided configuration","config_file":"/etc/caddy/Caddyfile","config_adapter":"caddyfile"}
{"level":"warn","ts":1706845343.9552839,"msg":"Caddyfile input is not formatted; run 'caddy fmt --overwrite' to fix inconsistencies","adapter":"caddyfile","file":"/etc/caddy/Caddyfile","line":16}
{"level":"info","ts":1706845343.956052,"logger":"admin","msg":"admin endpoint started","address":"localhost:2019","enforce_origin":false,"origins":["//localhost:2019","//[::1]:2019","//127.0.0.1:2019"]}
{"level":"info","ts":1706845343.95629,"logger":"tls.cache.maintenance","msg":"started background certificate maintenance","cache":"0xc000448580"}
{"level":"info","ts":1706845343.956486,"logger":"http.auto_https","msg":"server is listening only on the HTTPS port but has no TLS connection policies; adding one to enable TLS","server_name":"srv0","https_port":443}
{"level":"info","ts":1706845343.9565125,"logger":"http.auto_https","msg":"enabling automatic HTTP->HTTPS redirects","server_name":"srv0"}
{"level":"info","ts":1706845343.9588857,"msg":"FrankenPHP started 🐘","php_version":"8.3.2"}
{"level":"info","ts":1706845343.9607673,"logger":"tls","msg":"cleaning storage unit","storage":"FileStorage:/data/caddy"}
{"level":"info","ts":1706845343.9612813,"logger":"tls","msg":"finished cleaning storage units"}
{"level":"warn","ts":1706845343.976892,"logger":"pki.ca.local","msg":"installing root certificate (you might be prompted for password)","path":"storage:pki/authorities/local/root.crt"}
{"level":"info","ts":1706845343.9789531,"msg":"warning: \"certutil\" is not available, install \"certutil\" with \"apt install libnss3-tools\" or \"yum install nss-tools\" and try again"}
{"level":"info","ts":1706845343.9789617,"msg":"define JAVA_HOME environment variable to use the Java trust"}
{"level":"info","ts":1706845344.520868,"msg":"certificate installed properly in linux trusts"}
{"level":"info","ts":1706845344.521112,"logger":"http","msg":"enabling HTTP/3 listener","addr":":443"}
{"level":"info","ts":1706845344.5212042,"msg":"failed to sufficiently increase receive buffer size (was: 208 kiB, wanted: 2048 kiB, got: 416 kiB). See https://github.com/quic-go/quic-go/wiki/UDP-Buffer-Sizes for details."}
{"level":"info","ts":1706845344.5213459,"logger":"http.log","msg":"server running","name":"srv0","protocols":["h1","h2","h3"]}
{"level":"info","ts":1706845344.5214,"logger":"http.log","msg":"server running","name":"remaining_auto_https_redirects","protocols":["h1","h2","h3"]}
{"level":"info","ts":1706845344.521405,"logger":"http","msg":"enabling automatic TLS certificate management","domains":["localhost"]}
{"level":"info","ts":1706845344.5216856,"msg":"autosaved config (load with --resume flag)","file":"/config/caddy/autosave.json"}
{"level":"info","ts":1706845344.521702,"msg":"serving initial configuration"}
{"level":"info","ts":1706845344.5218856,"logger":"tls.obtain","msg":"acquiring lock","identifier":"localhost"}
{"level":"info","ts":1706845344.5223386,"logger":"tls.obtain","msg":"lock acquired","identifier":"localhost"}
{"level":"info","ts":1706845344.5224009,"logger":"tls.obtain","msg":"obtaining certificate","identifier":"localhost"}
{"level":"info","ts":1706845344.5233433,"logger":"tls.obtain","msg":"certificate obtained successfully","identifier":"localhost"}
{"level":"info","ts":1706845344.5234017,"logger":"tls.obtain","msg":"releasing lock","identifier":"localhost"}
{"level":"warn","ts":1706845344.5236707,"logger":"tls","msg":"stapling OCSP","error":"no OCSP stapling for [localhost]: no OCSP server specified in certificate","identifiers":["localhost"]}
{"level":"info","ts":1706845368.3628407,"logger":"http.log.access","msg":"handled request","request":{"remote_ip":"172.17.0.1","remote_port":"42776","client_ip":"172.17.0.1","proto":"HTTP/1.1","method":"GET","host":"127.0.0.1","uri":"/","headers":{"User-Agent":["curl/7.29.0"],"Accept":["*/*"]}},"bytes_read":0,"user_id":"","duration":0.000013394,"size":0,"status":308,"resp_headers":{"Content-Type":[],"Server":["Caddy"],"Connection":["close"],"Location":["https://127.0.0.1/"]}}
2024/02/02 03:42:59.801 INFO    http.log.access.log0    handled request {"request": {"remote_ip": "172.17.0.1", "remote_port": "42782", "client_ip": "172.17.0.1", "proto": "HTTP/1.1", "method": "GET", "host": "localhost", "uri": "/", "headers": {"User-Agent": ["curl/7.29.0"], "Accept": ["*/*"]}}, "bytes_read": 0, "user_id": "", "duration": 0.000009966, "size": 0, "status": 308, "resp_headers": {"Server": ["Caddy"], "Connection": ["close"], "Location": ["https://localhost/"], "Content-Type": []}}
```


```
[root@localhost ~]# curl localhost
[root@localhost ~]#

```

The request to localhost returns empty, and a static page should be returned

**Comment by kilwir76** [comment]
So : 

Create a new laravel project via composer .
Creates a docker-compose.yaml :
version: '3.9'

services:
    frankenphp:
        container_name: frankenphp-container
        build:
            context: .
            dockerfile: dockerfile
        ports:
            - 80:80
            - 443:443
            - 443:443/udp
        volumes:
            - $PWD:/app

    postgres:
        image: postgres:latest
        environment:
            POSTGRES_USER: frankenphp
            POSTGRES_PASSWORD: frankenphp
            POSTGRES_DB: frankenphp
        ports:
            - 5432:5432
        volumes:
            - frankenphp-db:/var/lib/postgresql/data

    adminer:
        image: adminer
        ports:
            - 8080:8080

volumes:
  frankenphp-db:

Creates a dockerfile.yaml :
FROM dunglas/frankenphp

# add additional extensions here:
RUN install-php-extensions \
	pdo_mysql \
    pdo_pgsql \
	gd
	intl
	zip
    bcmath
    opcache
    xdebug

COPY . /app/public

Then create an AuthController controller with an invoke method.
Add the passport package. Perform the classic migration to get the tables in the database.
Once you've done this, install . Retrieve credentials to connect via the url https://localhost/oauth/token with postman or insomnia.
And check that this doesn't work and that the frankenphp server is frozen. 

Basically, either I'm doing it wrong or I'm not having any problems with another FS project with a php-fpm docker and nginx classic. I also just noticed my potential error. I hadn't changed my APP_URL=http://localhost to "https". Now I get a return if I request the url https://localhost/oauth/token.
![Capture d'écran 2024-02-04 181141](https://github.com/dunglas/frankenphp/assets/13037707/68728fa2-a1bd-4e44-8b40-9589eed43969)
![Capture d'écran 2024-02-04 181249](https://github.com/dunglas/frankenphp/assets/13037707/7f9bd98b-313d-48f1-bf9b-640562be7145)

However, if I play this url in an invoke method, it really crashes the server without any output error.

CF : img

**Comment by kilwir76** [comment]
What I've just looked at, I could override the oauth/token uri via "AuthServiceProvider" but in any case what I've done via a magic method shouldn't crash the php server, in the worst case it should send me an error.

**Comment by dunglas** [comment]
This should definitely not crash the server. Could you provide a repository containing a minimal reproducer (it's very hard to deal with screenshots and incomplete reproducers)? This would allow us to get a stack trace and hopefully identify and squash the bug.

**Comment by kilwir76** [comment]
I do this for you asap. I'll put the rest link in this issue

**Comment by kilwir76** [comment]
Here is the repo in question :

https://github.com/MakFly/frankenphp-laravel

If you play with postman l'uri /api/oauth/token 0 answers emerge from this uri.

**Comment by d8vjork** [comment]
I'm having this exact same thing, I was trying this since this Laracon 2024 presentation but no luck to make it working

**Comment by withinboredom** [comment]
I'd be more than happy to take a gander, but so far, I don't have any way to reproduce the issue: just some logs and a repository, but no instructions for non-Laravel devs. For me, at least, I would need explicit commands to run without having to learn how Laravel works. As soon as I can reproduce, I will be hunting down the code running the code; Laravel isn't very important here, as that is most likely barely relevant if the server is crashing, I just need to be able to get the server into the right state.

**Comment by dunglas** [comment]
As I know Laravel, I'll be able to check when I'll be back to vacation, but a reproducer would definitely help!

**Comment by maxhelias** [comment]
@kilwir76 Can you try without installing xdebug?

**Comment by kilwir76** [comment]
ah yes i test tonight after my work.

**Comment by kilwir76** [comment]
well, problem solved. Obviously I didn't remember that opcache + xdebug weren't compatible together ...

Sorry for the inconvenience !

**Comment by kilwir76** [comment]
Basically, you can't have php opcache & xdebug enabled at the same time.
As far as I know, I had read somewhere that it was incompatible.
Here is my Dockerfile :

`FROM dunglas/frankenphp

# add additional extensions here:
RUN install-php-extensions \
	pdo_mysql \
    pdo_pgsql \
	gd \
	intl \
	zip \
    bcmath \
    # xdebug \
    opcache

COPY . /app/public
`
